_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,187 @@
+        end
+
+        function test_range_with_zero_stride_rhs_lt_lhs (~)
+            func = @()str_to_iarray ('3:0:-4');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        function test_range_with_zero_stride_rhs_gt_lhs (~)
+            func = @()str_to_iarray ('3:0:8');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        
+        %--------------------------------------------------------------------------
+        % non-Matlab format tokens
+        %--------------------------------------------------------------------------
+        function test_non_Matlab_range_pos2pos_ascend (~)
+            iarr = str_to_iarray ('3-8');
+            assertEqual (iarr, 3:8)
+        end
+        
+        function test_non_Matlab_range_pos2pos_descend (~)
+            iarr = str_to_iarray ('8-3');
+            assertEqual (iarr, 8:-1:3)
+        end
+        
+        function test_non_Matlab_range_neg2pos_ascend (~)
+            iarr = str_to_iarray ('-3-8');
+            assertEqual (iarr, -3:8)
+        end
+        
+        function test_non_Matlab_range_pos2neg_descend (~)
+            iarr = str_to_iarray ('3--8');
+            assertEqual (iarr, 3:-1:-8)
+        end
+        
+        function test_non_Matlab_range_neg2neg_ascend (~)
+            iarr = str_to_iarray ('-8--3');
+            assertEqual (iarr, -8:-3)
+        end
+        
+        function test_non_Matlab_range_neg2neg_descend (~)
+            iarr = str_to_iarray ('-3--8');
+            assertEqual (iarr, -3:-1:-8)
+        end
+       
+        function test_non_Matlab_range_neg2neg_single (~)
+            % Tough test of parsing?
+            iarr = str_to_iarray ('-3--3');
+            assertEqual (iarr, -3)
+        end
+       
+        function test_non_Matlab_range_neg2neg_single_FAIL (~)
+            % Tough test of parsing? Catch erroneous range indicator at front
+            func = @()str_to_iarray ('--3--3');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+       
+        
+        %--------------------------------------------------------------------------
+        % Tokens with whitespace
+        %--------------------------------------------------------------------------
+        function test_single_number_WS (~)
+            iarr = str_to_iarray ('  45  ');
+            assertEqual (iarr, 45)
+        end
+        
+        function test_range_with_positive_stride_rhs_gt_lhs_WS (~)
+            iarr = str_to_iarray ('  3:2:14  ');
+            assertEqual (iarr, 3:2:14)
+        end
+                
+        function test_range_with_positive_stride_rhs_gt_lhs_WSfail (~)
+            % Should fail as we do not permit whitespace within the token
+            func = @()str_to_iarray (' 3 :2: 14');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
```

Comment:
may be worth unpacking the message and checking the particular case

Reply:
Indeed. And the specific error message was not right. Code changed to make it correct

Reply:
syntax for message capture here is
`ME=assertExceptionThrown(func,'HERBERT:str_to_iarray:invalid_argument')`
from which ME.message will give you the error message rather than the generic identifier which is also the second argument.
assertEqual(ME.message,'whatever the message is') then checks

Reply:
I'd added a test on the specific error in my response to your previous comment; however I saw just now that I'd not done it for a couple of other tests that are expected to throw an error. I've now done added such tests.